src/templates/events.jsx

```diff
@@ -21,20 +21,50 @@ const Events = ({
     pageYear,
   },
 }) => {
-  const [upcomingEvents, setUpcomingEvents] = useState([]);
+  const [featuredUpcomingEvents, setFeaturedUpcomingEvents] = useState([]);
+  const [upcomingEventsByYear, setUpcomingEventsByYear] = useState({});
   const [pastEvents, setPastEvents] = useState([]);
-  const shouldShowUpcoming = pageYear === availableYears[0];
+  const currentYear = new Date().getFullYear();
+  const shouldShowUpcoming = pageYear >= currentYear;
 
   useEffect(() => {
-    setUpcomingEvents(
-      eventsGroupedByYears[availableYears[0]]
-        .filter((cEvent) => new Date(cEvent.acf.schedule.startDate) >= new Date()).reverse(),
-    );
+    const upcomingYears = Object.keys(eventsGroupedByYears)
+      .filter((key) => parseInt(key, 10) >= currentYear);
+
+    const getUpcomingEvents = (year) => {
+      let events = eventsGroupedByYears[year];
+      if (parseInt(year, 10) === currentYear) {
+        events = events
+          .filter((cEvent) => new Date(cEvent.acf.schedule.startDate) >= new Date());
+      }
+      const upcomingEvents = [...events].reverse();
+      return upcomingEvents;
+    };
```

Comment:
Let's make it pure and move it out of component body. Pass necessary data as parameters